File: backend/app/types/workflow.py
from pydantic import BaseModel
from typing import Optional


class WorkflowNodeSave(BaseModel):
	id: str
	node_type: str
	label: str
	approver_type: Optional[str] = None
	approver_id: Optional[int] = None
	position_x: float = 0
	position_y: float = 0


class WorkflowEdgeSave(BaseModel):
	id: str
	source: str
	target: str


class SaveGraphRequest(BaseModel):
	nodes: list[WorkflowNodeSave]
	edges: list[WorkflowEdgeSave]


class ToggleWorkflowRequest(BaseModel):
	is_active: bool

File: frontend/app/lib/types/workflow.ts
export interface WorkflowSummary {
	document_type: string;
	label: string;
	is_active: boolean;
}

export interface WorkflowNode {
	id: string;
	node_type: "start" | "end" | "approval";
	label: string;
	approver_type: "role" | "user" | null;
	approver_id: number | null;
	position_x: number;
	position_y: number;
}

export interface WorkflowEdge {
	id: string;
	source: string;
	target: string;
}

export interface WorkflowDetail extends WorkflowSummary {
	nodes: WorkflowNode[];
	edges: WorkflowEdge[];
}

export interface SaveGraphRequest {
	nodes: WorkflowNode[];
	edges: WorkflowEdge[];
}

export interface WorkflowHistoryStep {
	label: string;
	status: "approved" | "rejected" | "pending";
	actioned_by: string | null;
	actioned_at: string | null;
}

export interface WorkflowHistory {
	steps: WorkflowHistoryStep[];
}